src/test/java/com/cdancy/jenkins/rest/features/JobsApiLiveTest.java

```diff
@@ -332,6 +332,42 @@ public void testGetBuildCausesOfJob() {
         assertTrue(causes.get(0).userName().equals("admin"));
     }
 
+    public void testCreateJobForEmptyAndNullParams() {
+        String config = payloadFromResource("/freestyle-project-empty-and-null-params.xml");
+        RequestStatus success = api().create(null, "JobForEmptyAndNullParams", config);
+        assertTrue(success.value());
+    }
+
+    @Test(dependsOnMethods = "testCreateJobForEmptyAndNullParams")
+    public void testBuildWithParametersOfJobForEmptyAndNullParams() {
+        Map<String, List<String>> params = new HashMap<>();
+        params.put("SomeKey1", Lists.newArrayList(""));
+        params.put("SomeKey2", null);
+        IntegerResponse job1 = api.jobsApi().buildWithParameters(null, "JobForEmptyAndNullParams", params);
+        assertNotNull(job1);
+        assertTrue(job1.value() > 0);
+        assertTrue(job1.errors().size() == 0);
+    }
+
+    @Test(dependsOnMethods = "testBuildWithParametersOfJobForEmptyAndNullParams")
+    public void testGetBuildParametersOfJobForEmptyAndNullParams() {
+        while (api().jobInfo(null, "JobForEmptyAndNullParams").lastBuild() == null) {
```

Comment:
@richabindra if you look at the `QueueApiLiveTest` @martinda added a few tests around queuing builds that is more intuitive at the expense of a bit more code. I would use something like as this could potentially hang forever if someone is not careful. You may have to expose the method `getRunningQueueItem` and potentially make it static.

Reply:
sure.

Reply:
@richabindra what about moving the method to the `BaseJenkinsApiLiveTest` class?